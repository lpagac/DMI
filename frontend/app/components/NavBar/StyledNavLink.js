import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export default styled(NavLink)`
  font-size: 25px;
  margin: 0 40px;
  padding: 5px 5px;
  user-select: none;
  text-decoration: none;
  color: #fff;
  background-color: #bdb2ff;
  border-radius: 4px;

  &:hover {
    background-color: #7209b7;
  }
`;
