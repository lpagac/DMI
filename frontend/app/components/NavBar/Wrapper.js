import styled from 'styled-components';

const Wrapper = styled.nav`
  display: flex;
  justify-content: center;
  align-items: center;
  position: sticky;
  top: 0;
  min-height: 60px;
  background: #fff;
  border-bottom: 1px solid #dfe1e4;
`;

export default Wrapper;
