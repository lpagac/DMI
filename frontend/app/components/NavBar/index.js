/**
 *
 * NavBar
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';

import Wrapper from './Wrapper';
import StyledNavLink from './StyledNavLink';
import messages from './messages';

function NavBar() {
  const activeStyle = {
    backgroundColor: '#7209b7',
  };
  return (
    <Wrapper>
      <div>
        <StyledNavLink exact to="/strings" activeStyle={activeStyle}>
          <FormattedMessage {...messages.allStrings} />
        </StyledNavLink>
      </div>
      <div>
        <StyledNavLink exact to="/" activeStyle={activeStyle}>
          <FormattedMessage {...messages.home} />
        </StyledNavLink>
      </div>
      <div>
        <StyledNavLink exact to="/strings/add" activeStyle={activeStyle}>
          <FormattedMessage {...messages.addString} />
        </StyledNavLink>
      </div>
    </Wrapper>
  );
}

export default NavBar;
