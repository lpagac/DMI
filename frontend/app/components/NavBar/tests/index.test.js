/**
 *
 * Tests for NavBar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router';
import { createMemoryHistory } from 'history';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import NavBar from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from '../../../configureStore';

describe('<NavBar />', () => {
  let history;
  let store;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <NavBar />
          </IntlProvider>
        </ConnectedRouter>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <NavBar />
          </IntlProvider>
        </ConnectedRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
