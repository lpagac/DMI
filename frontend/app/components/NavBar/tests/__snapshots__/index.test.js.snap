// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Should render and match the snapshot 1`] = `
<nav
  class="Wrapper-sc-1tltayn-0 hQGFyS"
>
  <div>
    <a
      class="StyledNavLink-p1y3ih-0 VejsO"
      href="/strings"
    >
      <span>
        Strings
      </span>
    </a>
  </div>
  <div>
    <a
      aria-current="page"
      class="StyledNavLink-p1y3ih-0 VejsO active"
      href="/"
      style="background-color: rgb(114, 9, 183);"
    >
      <span>
        Home
      </span>
    </a>
  </div>
  <div>
    <a
      class="StyledNavLink-p1y3ih-0 VejsO"
      href="/strings/add"
    >
      <span>
        Add String
      </span>
    </a>
  </div>
</nav>
`;
